src/test/kotlin/no/nav/familie/ba/mottak/hendelser/LeesahConsumerTest.kt

```diff
@@ -0,0 +1,135 @@
+import no.nav.familie.prosessering.domene.TaskRepository
+import no.nav.person.pdl.leesah.Endringstype
+import no.nav.person.pdl.leesah.Personhendelse
+import no.nav.person.pdl.leesah.doedsfall.Doedsfall
+import no.nav.person.pdl.leesah.foedsel.Foedsel
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.generic.GenericRecordBuilder
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.Producer
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Tag
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.kafka.test.context.EmbeddedKafka
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.junit.jupiter.SpringExtension
+
+@SpringBootTest(properties = ["spring.kafka.bootstrap-servers=\${spring.embedded.kafka.brokers}"])
+@ExtendWith(SpringExtension::class)
+@ContextConfiguration(initializers = [DbContainerInitializer::class])
+@ActiveProfiles("postgres")
+@EmbeddedKafka(partitions = 1, topics = ["aapen-person-pdl-leesah-v1"])
+@Tag("integration")
+class LeesahConsumerTest {
+    @Autowired
+    lateinit var hendelsesloggRepository: HendelsesloggRepository
+
+    @Autowired
+    lateinit var kafkaProperties: KafkaProperties
+
+    @Autowired
+    lateinit var taskRepository: TaskRepository
+
+    private fun buildKafkaProducer(): Producer<Int, GenericRecord> {
+        val senderProps = kafkaProperties.buildProducerProperties()
+        return KafkaProducer(senderProps)
+    }
+
+    @Test
+    fun `Dødshendelse skal prosesseres uten feil`() {
+        val personhendelse = GenericRecordBuilder(Personhendelse.`SCHEMA$`)
+        personhendelse.set("hendelseId", "1")
+        val personidenter = ArrayList<String>()
+        personidenter.add("1234567890123")
+        personhendelse.set("personidenter", personidenter)
+        personhendelse.set("master", "")
+        personhendelse.set("opprettet", 0L)
+        personhendelse.set("opplysningstype", "DOEDSFALL_V1")
+        personhendelse.set("endringstype", Endringstype.OPPRETTET)
+
+        val dødsfall = GenericRecordBuilder(Doedsfall.`SCHEMA$`)
+        dødsfall.set("doedsdato", 1)
+        personhendelse.set("doedsfall", dødsfall.build())
+
+        val producer = buildKafkaProducer()
+        producer.send(ProducerRecord("aapen-person-pdl-leesah-v1", personhendelse.build()))
+        Thread.sleep(1000)
+    }
+
+    @Test
+    fun `Fødselshendelse skal prosesseres uten feil`() {
+        val personhendelse = GenericRecordBuilder(Personhendelse.`SCHEMA$`)
```

Comment:
Testene kan gjøres mer lesbar. F.eks. struktureres tydelig på given when then og egne metoder for å lage hendelser. Det er også manglende feiltester, og kun test på OPPRETTET/KORRIGERT